app/jobs/notify_creator_that_petition_is_published_email_job.rb

```diff
@@ -1,11 +1,22 @@
 class NotifyCreatorThatPetitionIsPublishedEmailJob < NotifyJob
-  self.template = :notify_creator_that_petition_is_published
+  include DateTimeHelper
+
+  def template_id(signature)
+    if signature.petition.collect_signatures?
+      self.template = :notify_creator_that_petition_is_published
+    else
+      self.template = :notify_creator_that_petition_is_published_without_signatures
+    end
```

Comment:
Did you try it the way I suggested first and it didn't work? I thought that it would inherit the class setting at the top and only override it for each specific job instance and not need resetting to the original value.